libvirt/resource_libvirt_network.go

```diff
@@ -105,6 +148,65 @@ func resourceLibvirtNetwork() *schema.Resource {
 	}
 }
 
+// a libvirt network DNS host template datasource
+//
+// Datasource example:
+//
+// data "libvirt_network_dns_host_template" "k8smasters" {
+//   count = "${var.master_count}"
+//   ip = "${var.master_ips[count.index]}"
+//   hostname = "master-${count.index}"
+// }
+//
+// resource "libvirt_network" "k8snet" {
+//   ...
+//   dns = [{
+//     hosts = [ "${flatten(data.libvirt_network_dns_host_template.k8smasters.*.rendered)}" ]
+//   }]
+//   ...
+// }
+//
+func datasourceLibvirtNetworkDNSHostTemplate() *schema.Resource {
```

Comment:
i am thinking that if we introduce a datasource resource, would make sense to me to create a new file 
as example:
https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/data_source_aws_acmpca_certificate_authority_test.go

https://github.com/terraform-providers/terraform-provider-google/tree/master/google